fleet-management-service/src/test/java/cz/fi/muni/pa165/facade/JourneyFacadeImplTest.java

```diff
@@ -0,0 +1,106 @@
+package cz.fi.muni.pa165.facade;
+
+import cz.fi.muni.pa165.config.ServiceConfiguration;
+import cz.fi.muni.pa165.entity.Employee;
+import cz.fi.muni.pa165.entity.Journey;
+import cz.fi.muni.pa165.entity.Vehicle;
+import cz.fi.muni.pa165.enums.Role;
+import cz.fi.muni.pa165.service.BeanMappingServiceImpl;
+import cz.fi.muni.pa165.service.JourneyServiceImpl;
+import cz.fi.muni.pa165.service.interfaces.BeanMappingService;
+import org.hibernate.service.spi.ServiceException;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import javax.inject.Inject;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.GregorianCalendar;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.*;
+
+/**
+ * @author Martin Schmidt
+ */
+@ContextConfiguration(classes = ServiceConfiguration.class)
+public class JourneyFacadeImplTest extends AbstractTestNGSpringContextTests {
+
+    @Spy
+    @Inject
+    private final BeanMappingService beanMappingService = new BeanMappingServiceImpl();
+
+    @InjectMocks
+    private final JourneyFacadeImpl journeyFacade = new JourneyFacadeImpl();
+
+    @Mock
+    private JourneyServiceImpl journeyService;
+
+    @BeforeClass
+    public void setup() throws ServiceException {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    private Date in;
+    private Date out;
+    private Employee employee;
+    private Vehicle vehicle;
+    private Journey journey;
+
+    @BeforeMethod
+    public void init() {
+        in = new GregorianCalendar(2016, 10, 10).getTime();
+        out = new GregorianCalendar(2016, 9, 1).getTime();
+        employee = new Employee("john.doe@muni.cz", "John", "Doe", "password", Role.EMPLOYEE);
+        vehicle = mock(Vehicle.class);
+        journey = new Journey(in, vehicle, employee);
+
+    }
+
+    @Test
+    public void testGetAllJourneys() {
+
+        when(journeyService.findAll()).thenReturn(Collections.singletonList(journey));
```

Comment:
You are working with entities here, you must work with DAOs! Service should also output just DAOs!